src/game_interpreter.cpp

```diff
@@ -4264,16 +4285,32 @@ bool Game_Interpreter::CommandManiacShowStringPicture(lcf::rpg::EventCommand con
 
 	text.font_size = font_px;
 
-	auto components = Utils::Tokenize(com.string, [](char32_t ch) {
-		return ch == '\x01';
+	// maniacs stores string parsing modes in the delimiters
+	// x01 -> literal string
+	// x02 -> direct reference
+	// x03 -> indirect reference
+	// for the displayed string, the id argument is in com.parameters[22]
+	// here we are capturing all the delimiters, but currently only need to support reading the first one
+	int i = 0;
+	int delims[3] = {};
```

Comment:
This will crash when there are for whatever reason more than 3 delimiters. Make this a vector or stop adding the modes when you are at 3.

Reply:
Oh good point -- consistency is not maniac's strong suit, haha

Reply:
✅